File: wallet/src/main/java/com/worldventures/wallet/analytics/settings/StealthModeAction.java
package com.worldventures.wallet.analytics.settings;

import com.worldventures.janet.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.wallet.analytics.WalletAnalyticsAction;

@AnalyticsEvent(action = "wallet:settings:privacy",
                trackers = AdobeTracker.TRACKER_KEY)
public class StealthModeAction extends WalletAnalyticsAction {

   @Attribute("privacystatus") final String privacyStatus;
   @Attribute("privacychange") String privacyChange = "1";

   public StealthModeAction(boolean stealthModeEnabled) {
      this.privacyStatus = stealthModeEnabled ? "Private" : "Not Private";
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/settings/StealthModeAction.kt
package com.worldventures.wallet.analytics.settings

import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.wallet.analytics.WalletAnalyticsAction

@AnalyticsEvent(action = "wallet:settings:privacy", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class StealthModeAction(stealthModeEnabled: Boolean) : WalletAnalyticsAction() {

   @Attribute("privacystatus") internal val privacyStatus: String = if (stealthModeEnabled) "Private" else "Not Private"
   @Attribute("privacychange") internal var privacyChange = "1"

}
